engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/date/ToEnsoDateNode.java

```diff
@@ -0,0 +1,33 @@
+package org.enso.interpreter.node.expression.builtin.date;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.data.EnsoDate;
+
+@BuiltinMethod(
+    type = "Date",
+    name = "internal_to_enso",
```

Comment:
I don't understand why we need this method. If the conversion happens automatically, why do you ever convert manually?

Reply:
Its only uses are wrong anyway, I'd say

Reply:
The way _Truffle/Host/Guest_ classes are being mixed in StdLib is very braided. Some might even call it wrong. The `Date.internal_to_enso` is just a way to get out of our current misery and start walking the right direction.

Reply:
My hope here would be to move away from using guest java for dates altogether in the stdlib. The reason you need this helper is using guest java _too much_. So we're probably in agreement w.r.t. future goals?